FT2-37-6: Completed the Basic Assignment 6.

MD-Aftab-Innoraft opened the pull request:
Description:
- Created a pdf displaying the details entered by user.

MD-Aftab-Innoraft committed:
FT2-37-6: Completed the Basic Assignment 6.
Description:
- Created a pdf displaying the details entered by user.

MD-Aftab-Innoraft committed:
FT2-37-6: Improved the appearance of generated pdf.
Description:
- Added gapping between cells to make the text appear better.
- Adjusted the image to fit in the desired space.

MD-Aftab-Innoraft committed:
FT2-37-6: Improved comments and code readability.
Description:
- Added comments stating parametrs and returns for methods.
- Specified input and output datatypes.
- Added proper spacing and alignment for better readability.

kumareshinnoraft commented on inputForm.php:
```diff
@@ -0,0 +1,367 @@
+<?php
+
+/* Form class to store the Form's information. */
```
Class comments are not proper as standards.

kumareshinnoraft commented on script.js:
```diff
@@ -0,0 +1,175 @@
+ *           (ii) string 'emailAddressRegex' to be validated against.
+ * @returns: string stating a relevant error (if any).
+ */
+
```
Extra spaces.

kumareshinnoraft commented on script.js:
```diff
@@ -0,0 +1,175 @@
+    return "*";
+}
+
+/**
+*   Function to check if the input Phone number is
+*   a valid Indian phone number.
+*   @params : (i)  string 'indianPhone' to be verified.
+*             (ii) string 'indianPhoneRegex' to be validated against.
+*   @returns : string containing relevant error message(if any).
+*   */ 
```
We should use proper PHP cs or Drupal coding standards.

kumareshinnoraft commented on pdfGenerate.php:
```diff
@@ -0,0 +1,182 @@
+    }
+}
+
+/**
+ * If the Form is submitted using POST request and 
+ * there are no errors in any input field,
+ * we create a pdf with the details. 
+ * One copy is saved on the server and 
+ * another copy is downloaded on the client's machine.
+ */
+if ($_SERVER["REQUEST_METHOD"] == "POST" && $myForm->checkNoError()) {
```
`if` `else` statements should have one line comment.

kumareshinnoraft commented on script.js:
```diff
@@ -0,0 +1,175 @@
+        errorFree = false;
+        return "* Only alphabets, white-spaces and ' allowed.";
+    }
+    else {
+        return "*";
+    }
```
No need of this else statement.

kumareshinnoraft commented on pdfGenerate.php:
```diff
@@ -0,0 +1,182 @@
+        $myPdf->Image("uploads/$img_name", 149, 37.8, 50, 51);
+    }
+
+    /** 
+     *   Generating the pdf file. 
+     */
```
Why we are adding multiline comments?

kumareshinnoraft commented on inputForm.php:
```diff
@@ -0,0 +1,367 @@
+      $data = json_decode($response, true);
+
+      if (
+        $data["deliverability"] != "DELIVERABLE" || $data["is_valid_format"]["text"] != "TRUE"
```
Why TRUE FALSE are used as string?

kumareshinnoraft commented on inputForm.php:
```diff
@@ -0,0 +1,367 @@
+    else {
+      $ch = curl_init();
+
+      curl_setopt($ch, CURLOPT_URL, 'https://emailvalidation.abstractapi.com/v1/?api_key=46cc4fa558dd4a17b2fbbd8402620c2e&email=' . $email);
+
+      curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+
+      $response = curl_exec($ch);
+
```
Why we are noting guzzle?

kumareshinnoraft commented on inputForm.php:
```diff
@@ -0,0 +1,367 @@
+    else if (!preg_match($phoneNumberRegex, $phoneNumber)) {
+      return "Only digits and + symbol allowed";
+    } 
+    else {
+      return "";
+    }
```
Unnecessary else statement.

kumareshinnoraft commented on inputForm.php:
```diff
@@ -0,0 +1,367 @@
+  /** 
+   *  Method to check if there are no error messages.
+   * 
+   *  @param  none
+   *   no input parameters.
```
If there is no parameter, we do not need to mention param in the comment.

kumareshinnoraft commented on inputForm.php:
```diff
@@ -0,0 +1,367 @@
+      return "Input length should be between 2 and 25";
+    } else if (!preg_match($regex, $data)) {
+      return "Invalid input";
+    } else {
+      return "";
```
Unnecessary else statement.

kumareshinnoraft reviewed (changes requested):
1. Follow proper coding standards.
2. Class name and file name should be same. And it should start with capital.
3. Create folders for different types of files.

MD-Aftab-Innoraft committed:
FT2-37-6: Followed PSR-2 coding standards.
Description:
- Changed the tab indent to 2 spaces.
- Followed proper formatting of if-else statements.
- Added proper comments using @var, @param and @return.
